hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/utils/TestWriteBase.java

```diff
@@ -137,14 +140,17 @@ public static TestHarness instance() {
     private String lastPending;
     private String lastComplete;
 
+    private HoodieFlinkWriteClient writeClient;
+
     public TestHarness preparePipeline(File basePath, Configuration conf) throws Exception {
       this.baseFile = basePath;
       this.basePath = this.baseFile.getAbsolutePath();
       this.conf = conf;
       this.pipeline = TestData.getWritePipeline(this.basePath, conf);
       // open the function and ingest data
       this.pipeline.openFunction();
-      this.ckpMetadata = CkpMetadata.getInstance(conf);
+      this.writeClient = FlinkWriteClients.createWriteClient(conf);
+      this.ckpMetadata = CkpMetadataFactory.getCkpMetadata(StreamerUtil.createMetaClient(conf), writeClient.getConfig(), conf);
```

Comment:
Not sure what is the purpose of the write config.

Reply:
`checkInflightInstant` and `lastPendingInstant` should request as a client to the server started by the coordinator, you can not set another server there and request from it, it works, but the workflow is not in line with the reality cases.